common/src/main/java/org/apache/rocketmq/common/UtilAll.java

```diff
@@ -83,15 +84,18 @@ public static int getPid() {
     }
 
     public static void sleep(long sleepMs) {
-        if (sleepMs < 0) {
+        sleep(sleepMs, TimeUnit.MILLISECONDS);
+    }
+
+    public static void sleep(long timeOut, TimeUnit timeUnit) {
+        if (null == timeUnit) {
             return;
         }
         try {
-            Thread.sleep(sleepMs);
-        } catch (Throwable ignored) {
+            timeUnit.sleep(timeOut);
+        } catch (InterruptedException ignored) {
```

Comment:
The new version is not equivalent to current version. IMO more relevant analysis is necessary to confirm this modification will not cause unexpected effects.

Reply:
@caigy 156
timeUnit.sleep(x)will call Thread.sleep after validating that the timeout is positive. Simplifies the previous code.
Other than that, the only difference is readability and using TimeUnitis probably easier to understand for non-obvious durations (for example: Thread.sleep(180000) vs. TimeUnit.MINUTES.sleep(3)). and I've adjusted the thrown exception to be Throwable.